demo-apps/cf-plugtest-client/src/main/java/org/eclipse/californium/plugtests/PlugtestClient.java

```diff
@@ -705,6 +719,103 @@ public static void testCL(String uri, EndpointContext context) throws ConnectorE
+
+		System.out.println("===============\nOSCORE 01");
+		System.out.println("---------------\nGET /test\n---------------");
+		Request r = Request.newGet();
+		r.getOptions().setOscore(Bytes.EMPTY);
+		response = client.advanced(r);
+		context = client.getDestinationContext();
+		if (context != null) {
+			System.out.println(Utils.prettyPrint(context));
+		}
+		System.out.println(response.advanced().getType() + "-" + response.getCode());
+		System.out.println(response.getResponseText());
+
+		System.out.println("===============\nOSCORE 02");
+		System.out.println("---------------\nDELETE /test\n---------------");
+		r = Request.newDelete();
+		r.getOptions().setOscore(Bytes.EMPTY);
+		response = client.advanced(r);
+		System.out.println(response.advanced().getType() + "-" + response.getCode());
+		System.out.println(response.getResponseText());
+
+		System.out.println("===============\nOSCORE 03");
+		System.out.println("---------------\nPUT /test\n---------------");
+		r = Request.newPut();
+		r.setPayload("");
+		r.getOptions().setContentFormat(MediaTypeRegistry.TEXT_PLAIN);
+		r.getOptions().setOscore(Bytes.EMPTY);
+		response = client.advanced(r);
+		System.out.println(response.advanced().getType() + "-" + response.getCode());
+		System.out.println(response.getResponseText());
+
+		System.out.println("===============\nOSCORE 04");
+		System.out.println("---------------\nPOST /test\n---------------");
+		r = Request.newPut();
+		r.setPayload("non-empty");
+		r.getOptions().setContentFormat(MediaTypeRegistry.TEXT_PLAIN);
+		r.getOptions().setOscore(Bytes.EMPTY);
+		response = client.advanced(r);
+		System.out.println(response.advanced().getType() + "-" + response.getCode());
+		System.out.println(response.getResponseText());
+		
+	}
+
+	/**
+	 * Initializes an OSCORE context for the client built on a pre-defined
+	 * configuration and adds it to the OSCORE context database. The created
+	 * context will support and by default always use the Appendix B.2 context
+	 * rederivation procedure. This is to avoid message replay errors that would
+	 * otherwise arise for clients.
+	 * 
+	 * @param uri the URI of the OSCORE resource at the server
+	 * @param db the OSCORE context database
+	 */
+	public static void initOscore(String uri, HashMapCtxDB db) {
+		AlgorithmID alg = AlgorithmID.AES_CCM_16_64_128;
+		AlgorithmID kdf = AlgorithmID.HKDF_HMAC_SHA_256;
+
+		byte[] master_secret = { 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E,
+				0x0F, 0x10 };
+		byte[] master_salt = { (byte) 0x9e, (byte) 0x7c, (byte) 0xa9, (byte) 0x22, (byte) 0x23, (byte) 0x78,
+				(byte) 0x63, (byte) 0x40 };
+		byte[] sid = new byte[] { 0x01 };
+		byte[] rid = new byte[] { 0x02 };
+		byte[] id_context = new byte[] { (byte) 0x37, (byte) 0xcb, (byte) 0xf3, (byte) 0x21, (byte) 0x00, (byte) 0x17,
+				(byte) 0xa2, (byte) 0xd3 };
+		int MAX_UNFRAGMENTED_SIZE = 4096;
```

Comment:
I'm not too sure, but `CoapConfig.MAX_RESOURCE_BODY_SIZE` is in an other place. We may also start to add a configuration module for oscore, and add oscore specific values to that.

Reply:
FMPOV, better than compiled in values.

Reply:
Now retrieving values from the standard configuration, modified in commit ee60caa4c866d635e36ee0e499433017246d7423